doc/api/parse_args.md

```diff
@@ -0,0 +1,60 @@
+# Parse arguments
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+<!-- source_link=lib/parse_args.js -->
+
+## `parseArgs([config])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* `config` {Object} Used to provide arguments for parsing and to configure
+  the parser. `config` supports the following properties:
+  * `args` {string\[]} array of argument strings. **Default:** `process.argv`
+    with `execPath` and `filename` removed.
+  * `options` {Object} Used to describe arguments known to the parser.
+    Keys of `options` are the long names of options and values are an
+    {Object} accepting the following properties:
+    * `type` {string} Type of argument, which must be either `boolean` or `string`.
+      **Default:** `boolean`.
+    * `multiple` {boolean} Should the option accept multiple arguments.
+    * `short` {string} A single character alias for the option.
+  * `strict`: {boolean} Should an error be thrown if unknown options are
```

Comment:
Is `strict` a common name for this behavior? I checked two modules on npm, and they both use it, but 'strict' doesn't seem like the most descriptive or future proof name.

Reply:
Assuming that calling parseArgs with `strict:false` never throws due to the end-user args, and `strict:true` checks for a number of potential end-user errors, do you have a suggestion for a better name? (I'm fairly used to `strict` and hadn't considered alternatives.)

Reply:
If the `strict` flag throws an error on unknown options only, as documented here, then it feels like 'strict' might be too broad of a term. Similar functionality from the joi module is named `allowUnknown`, but I don't feel super strongly about that.

> strict:true checks for a number of potential end-user errors

If `strict` is intended to perform a number of different checks, updating the docs to mention those other things might be the way forward instead.

Reply:
Yes, planning to go further than original description. (Thanks for explaining.)

Errors for:
- unknown option encountered
- option of `type:'string'` used like a boolean option e.g. lone `--string`
- option of `type:'boolean'` used like a string option e.g. `--boolean=bogus`

References for WIP:
- https://github.com/pkgjs/parseargs/issues/11#issuecomment-1082664767
- https://github.com/pkgjs/parseargs/pull/74

Reply:
I think many users of argument parsers will understand the meaning of strict. This is the terminology used by both commander and yargs, which represent a good chunk of the community.

I've updated the phrasing to I think better capture what `strict` mode represents:


```
Should an error be thrown if unknown and/or invalid options are encountered.
```